### ISSUE TITLE
Use Git commit time as file modification time

### ISSUE THREAD

**Issue by tewe** [glushchenko/fsnotes#1749]
### Description

I set up Git on iOS for the first time. There are files in there I haven't touched in 15 years. Yet they all show up as modified today. This is hiding valuable information from me, and the sort order doesn't match what I see on the Mac.

### To Reproduce

1. Fill out Git settings (for the first time)
2. Sort View by Modification Date

### FSNotes version

6.4.2

### macOS/iOS version

iOS 17.5.1

**Comment by glushchenko** [comment]
git does not store metadata

**Comment by tewe** [comment]
> git does not store metadata

You're right. It does store the time of each commit, and what files were changed. There are scripts to apply that to mtime after checkout. So please change this to a feature request. I know you can't currently work on those.

**Comment by glushchenko** [comment]
Experimentally in the latest release an option for storing dates in TextBundle has been added. Perhaps it will be a solution for someone.

**Comment by tewe** [comment]
Due to other sync issues I cannot test this right now, but [the commit](https://github.com/glushchenko/fsnotes/commit/fc8d8fff5a15144377fb7ebd980ed01b5edc51b0) suggests it works on iOS. Thanks!

It violates https://textbundle.org/spec/ because `created` and `modified` are top-level keys rather than in a `co.fluder.fsnotes` object.

**Comment by glushchenko** [comment]
I don't think there's any point in embedding this; if the modification dates are lost, they can easily be restored from git.

https://stackoverflow.com/questions/2458042/restore-a-files-modification-time-in-git

**Comment by tewe** [comment]
The issue I had was that, when sorting by modification time, I would see a different notes list on macOS and iOS. Because syncing via Git only preserves that metadata incidentally, and not for files last modified before setting up syncing.

**Comment by glushchenko** [comment]
@tewe what if we include two files in each commit, .ctime and .mtime, which will contain the time for all notes in the repository. And after we check out, show a dialog? “Restore file creation/modification time”? Then restore the time.

**Comment by tewe** [comment]
> [@tewe](https://github.com/tewe) what if we include two files in each commit, .ctime and .mtime, which will contain the time for all notes in the repository. And after we check out, show a dialog? “Restore file creation/modification time”? Then restore the time.

That would be more accurate than the commit time but clutter up the repo. It's also not trivial because of filename encoding and control characters. But I like the idea.

BTW I don't use FSNotes on iOS because it absolutely can't handle my 8700 notes.

**Comment by glushchenko** [comment]
Just two files in the root directory, which will contain utf-8 file names and the time for each of them.
